CONTRIBUTING.md

```diff
@@ -0,0 +1,60 @@
+# Contributing to MusicBrainz Server
+
+Thank you for considering contributing to MusicBrainz Server.
+
+Get started with https://musicbrainz.org/doc/Development first,
+continue with below additional guidelines then.
+
+## Submitting changes
+
+### Commit message
+
+We follow the generally accepted [seven rules of a great Git commit message
+](https://chris.beams.io/posts/git-commit/#seven-rules):
+
+1. Separate subject from body with a blank line
+1. Limit the subject line to 50 characters
+1. Capitalize the subject line
+1. Do not end the subject line with a period
+1. Use the imperative mood in the subject line
+1. Wrap the body at 72 characters
+1. Use the body to explain _what_ and _why_ vs. _how_
+
+Additionally, we start subject with ticket reference if applicable.
+
+### Pull request
+
+#### Ticket
+
+If your change is large or relevant to users, it should have a ticket, see
+[tickets](https://tickets.metabrainz.org/issues/?jql=project%20%3D%20MBS),
+create one if necessary.  It will be used to generate release notes.
+
+Untracked changes are typos, comments, coding style changes, automated
+dependency update, unnoticeable refactor, and so on.
+
+#### Title
+
+Describe your change with a short imperative title (like this)
+
+If your change relates to a ticket, see [above](#ticket), please make sure you
+prefix your pull request title with `MBS-XXX: ` in order for our
+ticket tracker to link your pull request to that ticket, e.g.
+
+> MBS-XXX: Title
+
+If your change has a ticket and if it completely resolves that ticket, please
+prefix title with either `Fix ` for a bugfix or `Implement ` for a ticket of
+some other type in order for our ticket tracker to transition that ticket, e.g.
+
+> Fix MBS-XXX: Title
+
+> Implement MBS-XXX: Title
+
+To sum up, title should follow of one of the these forms:
+
+* Change small unnoticeable bits
+* MBS-XXX: Change things relevant to users
+* Fix MBS-XXX: Completely resolve a reported bug
+* Implement MBS-XXX: Completely resolve a ticket of some other type
```

Comment:
Why?

It mimics [closing issues using keywords on GitHub](https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords) where keyword precedes issue number. It should be easy to parse for automation. It is easy to read and uses the same imperative mood as for commit messages (so that one-commit PR will have a correct title).

Alternative suggestions are welcome though.

Reply:
Just because the ticket number is IMO the most important part - it's assumed that you're fixing it or implementing it or whatever. But I can do that - I've been trying with the last PRs :)

Reply:
Ok, it makes sense to assume that pull request usually (or even should) completely resolve the mentioned ticket. So maybe, we can make the opposite: add keywords only for incomplete resolution? e.g. “MBS-XXX (Part Y): Did not anticipate that hassle"

Note: Automated transitions would be triggered by pull request title, not commit message.

Reply:
So the bot now supports parenthesis for partial resolution and still supports PRs’ title that starts as imperative sentence, but it no longer requires such keyword since it sets resolution to “Fixed” no matter ticket’s type, see OTHER-352.

In GitHub, we have label to distinguish bugfixes from other type of patches, so it isn’t very useful either. Plus, we were used to parenthesis already.

So I replaced the suggestion about keywords and replaced it with one about parenthesis in https://github.com/metabrainz/musicbrainz-server/pull/1345/commits/3fe58fc29b62ae96b1115f042aa8a026033bbd54#diff-6a3371457528722a734f3c51d9238c13R42-R60.